src/AltinnCore/Common/Clients/HttpClientAccessor.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using AltinnCore.Common.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace AltinnCore.Common.Clients
+{
+    /// <summary>
+    /// Http client accessor for accessing clients  for Altinn Platform integration
+    /// </summary>
+    public class HttpClientAccessor : IHttpClientAccessor
+    {
+        private readonly PlatformSettings _platformSettings;
+        private readonly HttpClient _storageClient;
+        private readonly HttpClient _registerClient;
+        private readonly HttpClient _profileClient;
+
+        /// <summary>
+        /// Initializes a new HttpClient accessor of the <see cref="HttpClientAccessor"/> class.
+        /// <param name="platformSettings">the platform settings</param>
+        /// </summary>
+        public HttpClientAccessor(IOptions<PlatformSettings> platformSettings)
+        {
+            _platformSettings = platformSettings.Value;
+
+            _storageClient = new HttpClient();
+            _storageClient.BaseAddress = new Uri($"{_platformSettings.GetApiStorageEndpoint}");
```

Comment:
So all the clients will be created when you call httpclientaccessor even though you might not need all of them all the time. Should the client creation be moved to the method for example.

Reply:
Smart, yeah I'll move it to the method.